### ISSUE TITLE
recording failed tests here that only happen on GitHub Actions. Odd thing is that these test pass on local checks

### ISSUE THREAD

**Issue by DanWismer** [NCC-CNC/wheretowork#363]
![image](https://github.com/NCC-CNC/wheretowork/assets/29556279/94553aa3-7a46-4090-ba12-4551c87e1a33)

**Comment by DanWismer** [comment]
![image](https://github.com/NCC-CNC/wheretowork/assets/29556279/f059a28d-f3a5-460a-aac5-de9f05469031)

**Comment by DanWismer** [comment]
![image](https://github.com/NCC-CNC/wheretowork/assets/29556279/af4cf01c-0179-469c-869b-66620258b1b3)

**Comment by DanWismer** [comment]
![image](https://github.com/NCC-CNC/wheretowork/assets/29556279/22ba9a8c-eb7f-4b21-9151-0a59f6e94374)

**Comment by jeffreyhanson** [comment]
Yeah that's weird. Is listed testthat in Suggests part of the DESCRIPTION?

**Comment by jeffreyhanson** [comment]
Hmm, also check if it's installed in renv?

**Comment by DanWismer** [comment]
Its not in the rene.lock file.

I tried renv::instal("testthat") and then renv::snapshot(), however it says the the lockfile is up to date. I think because it is not used in the application code the new version of renv can't pick it up?

We have it listed in the Suggests part of the Description

**Comment by jeffreyhanson** [comment]
Hmm, maybe try `renv::snapshot(type="explicit")` to capture those in the DESCRTION file?

**Comment by DanWismer** [comment]
I tried that, no dice.

I am reading this:
https://github.com/rstudio/renv/issues/1019

**Comment by DanWismer** [comment]
Maybe?
https://github.com/rstudio/renv/issues/315

**Comment by DanWismer** [comment]
so I did `renv::snapshot(dev=TRUE)` which will include the suggested packages into the renv.lock. 

However now, it says that the project is out of sync becuase the following packages are installed, recorded but not used...
![image](https://github.com/user-attachments/assets/de532f50-d0eb-4855-adea-a4e9b8d6ffe3)

**Comment by DanWismer** [comment]
@jeffreyhanson I am getting a weird faliure when it trys to install igraph:
![image](https://github.com/user-attachments/assets/be91366b-f2e5-4e18-a03b-a76085d21176)

I found out where it is called:

https://github.com/NCC-CNC/wheretowork/blob/316b2567ecc5a7d35572994490c34ec40d2403df/R/utils_spatial_misc.R#L89-L137

I am wondering if you have a new / different way to calculate reserve sizes, without the need to use `igraph`?

**Comment by jeffreyhanson** [comment]
Hmm, that's odd. Yeah, you could potentially use the sf package instead of igraph - but that will slow down performance a lot. I would suggest trying to work out why it's failing. It seems like igraph depends on cpp11, coudl you try updating that to the latest version?

**Comment by DanWismer** [comment]
Thanks, I see that cpp11 R package updated at the end of August. I updated to the latest 0.5 version and will see it that works.

**Comment by DanWismer** [comment]
Darn, still errors at the same spot. But only for Windows build

**Comment by jeffreyhanson** [comment]
Hmm, maybe the issue is one of its other dependencies?

**Comment by DanWismer** [comment]
I got it to pass on Windows!

in the `igraph 2.0.3` NEWS, GMP is no longer a dependency, and libxml2 and glpk are optional. 

I removed those from the `R-CMD-check-windows.yaml` script:
https://github.com/NCC-CNC/wheretowork/pull/380/commits/d6513acf105400e07d33d7c80f0e0b6005a2c785

I didn't see these system dependecies for Ubuntu or in the Dockerfile. @jeffreyhanson would you know why?

**Comment by jeffreyhanson** [comment]
Ah - nice work tracking that down! No idea why having those dependencies present would cause issues.

**Comment by DanWismer** [comment]
This still happens randomly:
![image](https://github.com/user-attachments/assets/c6de7918-79f0-4340-86a8-123d2a45bfab)

**Comment by DanWismer** [comment]
I think all these random fails have to do with the string encodings?

**Comment by DanWismer** [comment]
![image](https://github.com/user-attachments/assets/4e39ba1e-4d86-4ca3-99e1-8d48422d3407)
 can not subset columns that don't exists

**Comment by jeffreyhanson** [comment]
Yeah, that's odd - no idea what's going wrong here sorry. To help with debugging this, you could throw error messages that describe the contexts of `x`. E.g., something like: `stop(names(x))` or `stop(length(x))`
